[4.4 Regression] ia64: ICE: in instantiate_virtual_regs_in_insn, at function.c:1630
http://gcc.gnu.org/PR41551
https://bugs.gentoo.org/331825

--- branches/gcc-4_4-branch/gcc/function.c
+++ branches/gcc-4_4-branch/gcc/function.c
@@ -1601,7 +1601,13 @@
       if (!safe_insn_predicate (insn_code, i, x))
 	{
 	  start_sequence ();
-	  x = force_reg (insn_data[insn_code].operand[i].mode, x);
+	  if (REG_P (x))
+	    {
+	      gcc_assert (REGNO (x) <= LAST_VIRTUAL_REGISTER);
+	      x = copy_to_reg (x);
+	    }
+	  else
+	    x = force_reg (insn_data[insn_code].operand[i].mode, x);
 	  seq = get_insns ();
 	  end_sequence ();
 	  if (seq)
